torch/ao/quantization/fx/_model_report/model_report_observer.py

```diff
@@ -0,0 +1,77 @@
+import torch
+from torch.ao.quantization.observer import ObserverBase
+
+
+class ModelReportObserver(ObserverBase):
+    r"""This observer is used to record additional information regarding keeping track
+    of S = average_batch_activation_range/epoch_activation_range.
+
+    The purpose of this information is to prepare a report to present to users on whether
+    Dynamic or Static Quantization is more appropriate for their model given the general
+    distributions of their data.
+
+    Note: this tool is meant for FX Graph Mode Quantization
+    """
+
+    def __init__(self):
+        super().__init__(torch.qint8)
+        self.num_batches_tracked = 0
+
+        # keep track of the min and mix of the range for average batch and epoch as a whole
+        self.average_batch_activation_range = torch.tensor(float(0))
+        self.epoch_activation_min = torch.tensor(float("inf"))
+        self.epoch_activation_max = torch.tensor(float("-inf"))
+
+    def forward(self, x):
+        self.update_activation_range(x)
+        return x
+
+    def update_activation_range(self, x):
```

Comment:
nit: looks like this is only used once? if so, we can move everything to forward and delete this method I think